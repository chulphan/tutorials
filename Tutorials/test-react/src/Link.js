import React, { Component } from 'react';

class Link extends Component {
    state = {
        class: 'normal'
    }

    _onMouseEnter = () => {
        this.setState({
            class: 'hovered'
        })
    }

    _onMouseLeave = () => {
        this.setState({
            class: 'normal'
        })
    }

    render() {
        return (
            <a
                className={this.state.class}
                href={this.props.page || '#'}
                onMouseEnter={this._onMouseEnter}
                onMouseLeave={this._onMouseLeave}
            >
                {this.props.children}
            </a>
        )
    }
}

export default Link;