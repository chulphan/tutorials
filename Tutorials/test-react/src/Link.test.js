import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

// test (also under the alias 'it'.)
// : All you need in a test file is test method which runs a test.
// test(name, fn, timeout) 
// name: test name..(description), fn: contains the expectations to test. timeout: specifying how long to wait before aborting. (default: 5seconds.)

it('renders correctly', () => {
    const tree = renderer.create(<Link page="https://ekhub.co.kr">ekhub</Link>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders no page is set', () => {
    const tree = renderer.create(<Link>ekhub</Link>);

    expect(tree).toMatchSnapshot();
});

test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Link page="https://ekhub.co.kr">EKHUB</Link>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //manually trigger the callback
    tree.props.onMouseEnter();
    //re-rendereing
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //manually trigger the callback
    tree.props.onMouseLeave();
    //re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});