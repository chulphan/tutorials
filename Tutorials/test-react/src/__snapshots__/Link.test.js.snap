// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://ekhub.co.kr"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  EKHUB
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://ekhub.co.kr"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  EKHUB
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://ekhub.co.kr"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  EKHUB
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="https://ekhub.co.kr"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  ekhub
</a>
`;

exports[`renders no page is set 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  ekhub
</a>
`;
